components/modal/src/ModalHeader.js

```diff
@@ -0,0 +1,57 @@
+/** @jsx jsx */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { jsx, useTheme } from '@westpac/core';
+import { CloseIcon } from '@westpac/icon';
+import { Title } from './styled';
+
+// ==============================
+// Component
+// ==============================
+export const ModalHeader = ({ id, onClose, children, ...props }) => {
+	const { modal, colors } = useTheme();
+
+	return (
+		<div
+			css={{
+				display: 'flex',
+				justifyContent: 'space-between',
+				alignItems: 'flex-start',
+				overflow: 'hidden',
+				borderBottom: `${modal.header.borderWidth} solid ${modal.header.borderColor}`,
+				padding: modal.header.padding,
+			}}
+			{...props}
+		>
+			<Title id={id}>{children}</Title>
+			<button
+				css={{
+					border: 'none',
+					padding: '5px 12px',
+					cursor: 'pointer',
+					touchAction: 'manipulation',
+					marginTop: -10,
+					marginRight: -22,
+
+					'&:hover svg': { opacity: 0.5 },
+				}}
+				onClick={onClose}
+			>
+				<CloseIcon size="small" color={colors.text} />
+			</button>
+		</div>
+	);
+};
+
+ModalHeader.propTypes = {
+	/** id of the title for accessibility */
+	id: PropTypes.string,
```

Comment:
.isRequired?

Reply:
I already handle the aria-describedby and aria-labelledby by auto-generating an id an passing to appropriate modal components